### ISSUE TITLE
Server URL in client and unattended agent defaults to http even with ca certified LetsEcrypt https setup

### ISSUE THREAD

**Issue by bmcgonag** [immense/Remotely#584]
**Describe the bug**
I installed remotely using Docker. 
I setup my reverse proxy and it is working.
I used LetsEncrypt to get proper https working.
I download the client from my site, and run it on Windows 10 or 11, and the server URL is defaulted to http instead of https.

Where does the server URL get set?  Can I set this so it will default to https for future downloads of the client or agent?

If not, could this be exposed as an Environment Variable to set in Docker?

**To Reproduce**
1. Install the server.
2. Setup a reverse proxy with letsencrypt certificates for CA https access, and force https only.
3. Download the client (or agent) and run it on Windows 10 or 11. The app will fail to connect because the server url is set to 'http' instead of 'https'.

**Remotely Version**
Server Version: 2023.01.20.0732
Agent Version: 2023.01.20.0732

**Expected Behavior**
The client or agent, when initially launched (set for install) should have the proper connection type for the server (preferably defaulting to https if unknown).

**Desktop (please complete the following information):**
 - OS: Windows 10  Pro 64 Bit, Windows 11 Pro 64 Bit
 - Browser: Firefox 109.0.1, Chrome latest, Chromium Latest

**Smartphone (please complete the following information):**
NA

**Additional Context**
Love the software, just want to know how to resolve this.

**Comment by lucent-sea** [comment]
The server URL is determined from the properties of the incoming request.  The scheme (http/https) is obtained from the `X-Forwarded-Proto` header, using ASP.NET Core's [forwarded headers middleware](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer#forwarded-headers). 

The parsing and modification of the request is done internally within ASP.NET Core's code.  Remotely just uses the resulting values.

If the scheme isn't correct, it means that the header is missing, or the chain of proxies aren't trusted.  I explained more of it here: https://github.com/immense/Remotely/issues/573

I'm using the Host header directly instead of `X-Forwarded-Host` because proxies don't usually modify that header.  So the host can still be correct even if the forwarded headers aren't trusted (and thus not used).

However, I ended up creating a "Force Client HTTPS" option in the Server Config page (i.e. appsettings.json) to work around this.
